### PR TITLE
Starting to add some UI improvements from HD mod (dead code)

### PR DESCRIPTION
**Pull request by ars3niy**
Display some new buttons (doing nothing) like in the HD mod in hero exchange dialog when certain additional resource file are present. Those resource files could later be added into VCMI essential add-on, but since no one is going to have them now, all this is currently dead code.

### CODE REVIEW

File: client/widgets/Images.cpp [hunk 1]
```diff
@@ -259,10 +290,26 @@ void CAnimImage::showAll(SDL_Surface * to)
 
 	if(flags & CShowableAnim::BASE && frame != 0)
 		if(auto img = anim->getImage(0, group))
-			img->draw(to, pos.x, pos.y);
+		{
```
Comment (alexvins) [inline client/widgets/Images.cpp:293]: Put `{}` at top level

File: client/widgets/Images.cpp [hunk 2]
```diff
@@ -259,10 +290,26 @@ void CAnimImage::showAll(SDL_Surface * to)
...
+			if (isScaled())
+			{
+				auto scaled = img->scaleFast(float(scaledSize.x) / img->width());
```
Comment (alexvins) [inline client/widgets/Images.cpp:296]: Scaled image should be cached

File: client/windows/GUIClasses.cpp [hunk 3]
```diff
@@ -787,9 +787,18 @@ void CTavernWindow::HeroPortrait::hover(bool on)
 		GH.statusbar->clear();
 }
 
+static const char *NEW_EXCHANGE_BG = "TRADE2X";
```
Comment (alexvins) [inline client/windows/GUIClasses.cpp:790]: "NewSmth" is not a good name, these features are new up until this PR is merged.

File: client/widgets/Images.cpp (hunk 2 above)
Comment (ars3niy) [reply client/widgets/Images.cpp:296]: Maybe I'll just add new resource file with the new icon size instead. This dialog needs a bunch of new icons anyway.

**Comment by SourceCodeDeleted** [comment]
Does this fix the UI screen problems on the phone? Where the screen isn't stretched enough for the entire screen?

**Comment by nullkiller** [comment]
Good job @ArseniyShestakov , picked your changes and sent another PR

**Comment by nullkiller** [comment]
Outdated

**Pull request closed without merging**